build fail implicit declaration of function 'av_set_saf_open'

**Description**
```
DEBUG: Copied the license file of freetype successfully
DEBUG: Copied the license file of x264 successfully
DEBUG: Copied the license file of libpng successfully
DEBUG: Copied the license file of cpu_features successfully
DEBUG: Copied the ffmpeg-kit license successfully

[armeabi-v7a] Compile arm    : ffmpegkit <= ffmpegkit.c
jni/../ffmpeg-kit-android-lib/src/main/cpp/ffmpegkit.c:665:5: error: implicit declaration of function 'av_set_saf_open' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    av_set_saf_open(saf_open);
    ^
jni/../ffmpeg-kit-android-lib/src/main/cpp/ffmpegkit.c:666:5: error: implicit declaration of function 'av_set_saf_close' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    av_set_saf_close(saf_close);
    ^
jni/../ffmpeg-kit-android-lib/src/main/cpp/ffmpegkit.c:666:5: note: did you mean 'av_set_saf_open'?
jni/../ffmpeg-kit-android-lib/src/main/cpp/ffmpegkit.c:665:5: note: 'av_set_saf_open' declared here
    av_set_saf_open(saf_open);
    ^
2 errors generated.
```

**Expected behavior**
no error

**Current behavior**
build fail

**To Reproduce**

```
commit f8c5b16bdb4b3d24a71deb74307717864282d66d (HEAD -> main, origin/main, origin/HEAD)

./android.sh --lts --api-level=21 --enable-x264 --enable-gpl --disable-x86 --disable-x86-64 --disable-arm64-v8a --no-ffmpeg-kit-protocols --enable-freetype
```


**Logs**
Post logs here or paste them to [Ghostbin](https://ghostbin.co) and insert the link here.

**Environment**

- Platform: `Android`
- Architecture: `arm-v7a`, 
- Version: `v4.5.LTS`
- Source branch: `main`
- Xcode version: `12.4.1`
- Cocoapods version: `1.10.1`
- Android Studio version: `4.0`
- Android NDK version: `r22b`
- flutter doctor: ` `
- react-native info: ` `

**Other**
Add any other context about the problem here.

--- a/.github/workflows/android-build-scripts.yml
+++ b/.github/workflows/android-build-scripts.yml
@@ -132,3 +132,32 @@ jobs:
       - name: print ffbuild logs
         if: ${{ failure() }}
         run: '[[ -f ./src/ffmpeg/ffbuild/config.log ]] && tail -50 ./src/ffmpeg/ffbuild/config.log'
+  build-main-without-ffmpeg-kit-protocols-on-linux:
+    name: build without ffmpeg-kit protocols
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        ndk-version: [ 'r22b-linux-x86_64' ]
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - uses: actions/checkout@v2
+      - name: set up adopt jdk 8
+        uses: actions/setup-java@v2.2.0
+        with:
+          distribution: 'adopt'
+          java-version: '8'
+      - name: set up android ndk
+        run: |
+          curl -s "https://dl.google.com/android/repository/android-ndk-${{ matrix.ndk-version }}.zip" -o ndk.zip
+          unzip -q -o ndk.zip -d .ndk
+          echo "ANDROID_NDK_ROOT=$PWD/.ndk/$(ls .ndk)" >> $GITHUB_ENV
+      - name: run the build script
+        run: ./android.sh --no-ffmpeg-kit-protocols --disable-x86 --disable-x86-64 --disable-arm-v7a --disable-arm-v7a-neon
+      - name: print build logs
+        if: ${{ always() }}
+        run: cat build.log
+      - name: print ffbuild logs
+        if: ${{ failure() }}
+        run: '[[ -f ./src/ffmpeg/ffbuild/config.log ]] && tail -50 ./src/ffmpeg/ffbuild/config.log'

--- a/scripts/android/ffmpeg.sh
+++ b/scripts/android/ffmpeg.sh
@@ -405,7 +405,7 @@ ${SED_INLINE} "s/\$version/$FFMPEG_VERSION/g" "${BASEDIR}"/src/"${LIB_NAME}"/ffb
 
 # 3. Enable ffmpeg-kit protocols
 if [[ ${NO_FFMPEG_KIT_PROTOCOLS} == "1" ]]; then
-  ${SED_INLINE} "s/ av_set_fd_close/\/\/av_set_fd_close/g" "${BASEDIR}"/android/ffmpeg-kit-android-lib/src/main/cpp/ffmpegkit.c 1>>"${BASEDIR}"/build.log 2>&1
+  ${SED_INLINE} "s| av_set_saf|//av_set_saf|g" "${BASEDIR}"/android/ffmpeg-kit-android-lib/src/main/cpp/ffmpegkit.c 1>>"${BASEDIR}"/build.log 2>&1
   echo -e "\nINFO: Disabled custom ffmpeg-kit protocols\n" 1>>"${BASEDIR}"/build.log 2>&1
 else
   cat ../../tools/protocols/libavformat_file.c >> libavformat/file.c

